### ISSUE TITLE
ps-active-y not applied until initial scroll on container element

### ISSUE THREAD

**Issue by mattmarcello** [noraesae/perfect-scrollbar#416]
I am using perfect-scrollbar in the context of a React component.  After a scroll event on the container element, it works as expected, but not until then.  

Has anyone encountered this issue?

I initialize the Scrollbar as follows
```js
...

  componentDidMount() {
    let container = ReactDOM.findDOMNode(this);
    Ps.initialize(container);
  }

...

```

Best, 
Matt

**Comment by DanielApt** [comment]
I have noticed this as well. The scrollbar is only rendered once scrolling is called.

I'm not sure, but could you try using `Ps.update(container)` once the container has mounted?

**Comment by noraesae** [comment]
This is likely to be a timing issue. I'm not sure the React component fills the content dynamically after componentDidMount. If it does, `Ps.update` should be called. Could you try calling the method in `componentDidUpdate`?

**Comment by MartijnBroere** [comment]
I am having sort of the same issue. When the page is loaded, i want the element on which perfect-scrollbar is applied to (`container`), to be horizontally scrolled to a given position. This is done by `container.animate({scrollLeft:givenPosition})`. After this perfect-scrollbar works perfectly fine, just as expected. 

But on a touch-device (tested on iPad Air on iOs Simulator) the first touch after page-load makes the `container` jump to the ultimate left, i am assuming scrolled to 0. If i `touchmove` after this, it works again just fine.

Could this issue also be related to #414 ?

**Comment by BKD-Digital** [comment]
I needed to quickly resolve this issue for us on a project. Since it is a requirement to always display scrollbars we created the following CSS rule (Sass rule):

  .parent-container {
    &.ps-container > .ps-scrollbar-x-rail,
    &.ps-container > .ps-scrollbar-y-rail {
      .ps-scrollbar-y {
        min-height: 90px;
      }
      opacity: .6;
      display: block;
      min-height: 100px;
    }
}

**Comment by ciccia** [comment]
Same for me. I'm on an Angular project. 
I fixed it calling (in a digest apply) update wrapped in a timeout.

```javascript
$timeout(function () {
          $element.find('.table-responsive').perfectScrollbar('update');
        }, 100, false);
```

**Comment by noraesae** [comment]
I still think that it may have been a timing issue. This can be solved by calling Ps.update when it is actually needed, or calling it ad hoc with timeout.

I close this issue here, but please feel free to reopen it or open a new issue if having further problem.

Thanks!
